WordPress/Classes/ViewRelated/Site Creation/FinalAssembly/SiteCreationRotatingMessageView.swift

```diff
@@ -0,0 +1,189 @@
+
+        static let horizontalSpacing: CGFloat = 15
+    }
+
+    private struct Parameters {
+        //Animation parameters
+        static let transitionAnimationDuration = TimeInterval(0.5)
+        private static let statusDisplayDuration = TimeInterval(2)
+
+        //Adjust the total display time to take the fade into consideration
+        static var totalStatusDisplayDuration: TimeInterval {
+            return statusDisplayDuration + transitionAnimationDuration
+        }
+    }
+
+    /// An array of status messages to rotating through
+    private(set) var statusMessages: [String]!
+
+    // MARK: - State Management
+
+    /// A timer that determines how long to show each message
+    internal var animationTimer: Timer!
+
+    /// The index of the currently visible message
+    internal var visibleIndex = 0
+
+    /// Calculates the index that should be displayed next,
+    /// This will automatically loop back to 0 if we hit the end
+    internal var nextVisibleIndex: Int {
+        get {
+            var nextIndex = visibleIndex + 1
+            if nextIndex >= statusMessages.count {
+                nextIndex = 0
+            }
+
+            return nextIndex
+        }
+    }
+
+    // MARK: - View Properties
+    private var statusImageView: UIImageView!
+
+    /// This advises the user that the site creation request is underway.
+    private(set) var statusLabel: UILabel = {
+        let label = UILabel()
+
+        label.translatesAutoresizingMaskIntoConstraints = false
+
+        label.font = WPStyleGuide.fontForTextStyle(.title2)
+        label.textColor = .textSubtle
+        label.textAlignment = .center
+
+        return label
+    }()
+
+    /// The stack view manages the appearance of a icon image, and a status label
+    private let statusStackView: UIStackView = {
+        let stackView = UIStackView()
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.alignment = .center
+        stackView.axis = .horizontal
+        stackView.distribution = .fillProportionally
+        stackView.alignment = .center
+        stackView.spacing = Margins.horizontalSpacing
+
+        return stackView
+    }()
+
+    // MARK: - UIView
+
+    /// Creates a new instance of the message view with the provided messages
+    /// - Parameter messages: An array of messages to display
+    /// - Parameter iconImage: The icon image to display before the status message
+    init(messages: [String], iconImage: UIImage) {
+        self.statusMessages = messages
+
+        self.statusImageView = UIImageView(image: iconImage)
+
+        super.init(frame: .zero)
+        configure()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Public Methods
+    func startAnimating() {
+        stopAnimating()
+
+        self.animationTimer = Timer.scheduledTimer(timeInterval: Parameters.totalStatusDisplayDuration,
```

Comment:
We can remove the `self`.

```suggestion
       animationTimer = Timer.scheduledTimer(timeInterval: Parameters.totalStatusDisplayDuration,
```

There are other `self` that we could remove without any side effect,

Reply:
Oops, habits from writing Objective-C